sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -612,29 +615,24 @@ private static boolean isPortResource(Protos.Resource resource) {
         Protos.NetworkInfo.Builder netInfoBuilder = Protos.NetworkInfo.newBuilder();
         netInfoBuilder.setName(networkSpec.getName());
 
-        if (!networkSpec.getPortMappings().isEmpty()) {
+        if (!networkSpec.getPortMappings().isEmpty() &&
+                DcosConstants.networkSupportsPortMapping(networkSpec.getName())) {
+            // we double check the availability of port mapping here in case the service spec was made in
+            // pure java instead of YAML
             for (Map.Entry<Integer, Integer> e : networkSpec.getPortMappings().entrySet()) {
                 Integer hostPort = e.getKey();
                 Integer containerPort = e.getValue();
+                LOGGER.info("Mapping container port {} to host port {}", containerPort, hostPort);
                 netInfoBuilder.addPortMappings(Protos.NetworkInfo.PortMapping.newBuilder()
                         .setHostPort(hostPort)
                         .setContainerPort(containerPort)
+                        .setProtocol("tcp")  // TODO(arand) check that this is necessary
                         .build());
             }
         }
 
-        if (!networkSpec.getNetgroups().isEmpty()) {
-            netInfoBuilder.addAllGroups(networkSpec.getNetgroups());
-        }
-
-        if (!networkSpec.getIpAddresses().isEmpty()) {
-            for (String ipAddressString : networkSpec.getIpAddresses()) {
-                netInfoBuilder.addIpAddresses(
-                        Protos.NetworkInfo.IPAddress.newBuilder()
-                                .setIpAddress(ipAddressString)
-                                .setProtocol(Protos.NetworkInfo.Protocol.IPv4)
-                                .build());
-            }
+        if (!networkSpec.getLabels().isEmpty()) {
```

Comment:
This is coming very soon and is already plumbed in the rest of the interfaces. If you feel strongly that we shouldn't have any partially implemented code, I can remove it from `[raw]NetworkSpec` and `DefaultNetworkSpec`.